datastore/badger/application.go

```diff
@@ -23,10 +23,28 @@ func NewApplicationRepo(db *badgerhold.Store) datastore.ApplicationRepository {
 }
 
 func (a *appRepo) CreateApplication(ctx context.Context, app *datastore.Application) error {
+	unique, err := a.isAppTitleUnique(ctx, app)
```

Comment:
@danvixent Change `func isAppTitleUnique(ctx, app) (bool, error)` to `func assertUniqueAppTitle(ctx, app) error` let it return errors or `datastore.ErrDuplicateAppName` and remove the duplicate checks here.